AsyncDisplayKit/ASCellNode.mm

```diff
@@ -117,15 +118,13 @@ - (void)layoutDidFinish
   _viewControllerNode.frame = self.bounds;
 }
 
-- (void)__setNeedsLayout
+- (void)displayNodeDidInvalidateSizeNewSize:(CGSize)newSize
 {
-  CGSize oldSize = self.calculatedSize;
-  [super __setNeedsLayout];
-  
-  //Adding this lock because lock used to be held when this method was called. Not sure if it's necessary for
-  //didRelayoutFromOldSize:toNewSize:
-  ASDN::MutexLocker l(__instanceLock__);
-  [self didRelayoutFromOldSize:oldSize toNewSize:self.calculatedSize];
+  CGSize oldSize = self.bounds.size;
+  if (CGSizeEqualToSize(oldSize, newSize) == NO) {
+    self.frame = {self.frame.origin, newSize};
+    [self didRelayoutFromOldSize:oldSize toNewSize:newSize];
+  }
```

Comment:
I renamed `displayNodeDidInvalidateSizeNewSize` to `_locked_ displayNodeDidInvalidateSizeNewSize`. We use it internally for now in `ASDisplayNode` and `ASCellNode` and in both times we would have get the lock in there so let's push it up a bit. If we would like to open this API we have to re-think how to communicate what the best strategy would be.